frontend/src/app/core/interceptors/jwt.interceptor.ts

```diff
@@ -0,0 +1,27 @@
+import {
+    HttpEvent,
+    HttpHandler,
+    HttpInterceptor,
+    HttpRequest } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { FirebaseAuthService } from '@core/services/firebase-auth.service';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class JwtInterceptor implements HttpInterceptor {
+    constructor(private firebaseAuthService: FirebaseAuthService) {}
+
+    intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+        const userToken = this.firebaseAuthService.getUserToken();
```

Comment:
we need to use Auth service here and take a token from local storage (not load every time from Firebase)

if the token does not exists in local storage then load from firebase

Reply:
done